### ISSUE TITLE
CLI with --lr_scheduler does not work if optimizer is configured via configure_optimizers

### ISSUE THREAD

**Issue by lendle** [Lightning-AI/pytorch-lightning#17800]
### Bug description

Per the example code snippet in the [docs](https://lightning.ai/docs/pytorch/stable/cli/lightning_cli_intermediate_2.html#multiple-schedulers), you should be able to configure an optimizer in `configure_optimizers` and then specify a LR scheduler via cli argument `--lr_scheduler`. 

The code snippet is 
```
# main.py
import torch
from lightning.pytorch.cli import LightningCLI
from lightning.pytorch.demos.boring_classes import DemoModel, BoringDataModule


class LitLRScheduler(torch.optim.lr_scheduler.CosineAnnealingLR):
    def step(self):
        print("⚡", "using LitLRScheduler", "⚡")
        super().step()


cli = LightningCLI(DemoModel, BoringDataModule)
```

DemoModel returns an optimizer via `configure_optimizers`. 

The custom LitLRScheduler does not get run via `python main.py fit --lr_scheduler LitLRScheduler --lr_scheduler.T_max 100  --trainer.accelerator cpu` as implied by the docs. ("⚡", "using LitLRScheduler", "⚡" is never printed).

Note I had to include `--lr_scheduler.T_max` because T_max is a required parameter of `CosineAnnealingLR`, so the example wouldn't even run without error out of the box. 

### What version are you seeing the problem on?

v2.0

### Error messages and logs

```
# Error messages and logs here please
```

### Environment

<details>
  <summary>Current environment</summary>

```
#- Lightning Component (e.g. Trainer, LightningModule, LightningApp, LightningWork, LightningFlow): LightningModule, CLI
#- PyTorch Lightning Version (e.g., 1.5.0): commit dea1ff6 -- I have to pin this version because I need the behavior of bf16-mixed w/ fsdp which changed after this commit, and bf16-true is not available yet for fsdp.
#- Lightning App Version (e.g., 0.5.2):
#- PyTorch Version (e.g., 2.0): 2.0.1
#- Python version (e.g., 3.9): 3.10.10
#- OS (e.g., Linux): linux
#- CUDA/cuDNN version:
#- GPU models and configuration:
#- How you installed Lightning(`conda`, `pip`, source): pip
#- Running environment of LightningApp (e.g. local, cloud):

```

</details>

cc @carmocca @mauvilsa

**Comment by carmocca** [comment]
We recently updated the docs about this (https://github.com/Lightning-AI/lightning/pull/17552). It requires that you also specify the `--optimizer`

**Comment by carmocca** [comment]
In that case, passing an optimizer or optimizer and scheduler will override the model's `configure_optimizers`

**Comment by awaelchli** [comment]
@lendle Were you able to make it work?
